pubsub/google/cloud/pubsub_v1/subscriber/client.py

```diff
@@ -0,0 +1,100 @@
+            settings. Use this to prevent situations where you are
+            inundated with too many messages at once.
+        policy_class (class): A class that describes how to handle
+            subscriptions. You may subclass the
+            :class:`.pubsub_v1.subscriber.policy.base.BasePolicy`
+            class in order to define your own consumer. This is primarily
+            provided to allow use of different concurrency models; the default
+            is based on :class:`threading.Thread`.
+        kwargs (dict): Any additional arguments provided are sent as keyword
+            keyword arguments to the underlying
+            :class:`~.gapic.pubsub.v1.subscriber_client.SubscriberClient`.
+            Generally, you should not need to set additional keyword
+            arguments.
+    """
+    def __init__(self, flow_control=(), policy_class=thread.Policy,
+                 **kwargs):
+        # Add the metrics headers, and instantiate the underlying GAPIC
+        # client.
+        kwargs['lib_name'] = 'gccl'
+        kwargs['lib_version'] = __VERSION__
+        self.api = subscriber_client.SubscriberClient(**kwargs)
+
+        # The subcription class is responsible to retrieving and dispatching
+        # messages.
+        self._policy_class = policy_class
+
+    def subscribe(self, subscription, callback=None):
+        """Return a representation of an individual subscription.
+
+        This method creates and returns a ``Consumer`` object (that is, a
+        :class:`~.pubsub_v1.subscriber.consumer.base.BaseConsumer`)
+        subclass) bound to the topic. It does `not` create the subcription
+        on the backend (or do any API call at all); it simply returns an
+        object capable of doing these things.
+
+        If the ``callback`` argument is provided, then the :meth:`open` method
+        is automatically called on the returned object. If ``callback`` is
+        not provided, the subscription is returned unopened.
+
+        .. note::
+            It only makes sense to provide ``callback`` here if you have
+            already created the subscription manually in the API.
+
+        Args:
+            subscription (str): The name of the subscription. The
+                subscription should have already been created (for example,
+                by using :meth:`create_subscription`).
+            callback (function): The callback function. This function receives
+                the :class:`~.pubsub_v1.types.PubsubMessage` as its only
+                argument.
+            flow_control (~.pubsub_v1.types.FlowControl): The flow control
+                settings. Use this to prevent situations where you are
+                inundated with too many messages at once.
+
+        Returns:
+            ~.pubsub_v1.subscriber.consumer.base.BaseConsumer: An instance
+                of the defined ``consumer_class`` on the client.
+        """
+        subscr = self._policy_class(self, subscription)
```

Comment:
`policy` instead of `subscr`? (you're constructing an instance of the policy class, so it leads that the name should match).